### ISSUE TITLE
Hiding apps inconsistency.

### ISSUE THREAD

**Issue by SoPat712** [lwouis/alt-tab-macos#567]
**Describe the bug**

When hiding an app in Alt-Tab, the icon shows up next to the app, and then the app is sent to the end of the list. I think that sending it instantly to the end of the list is confusing with tons of apps open. Sometimes I just want to go through all my apps and hide some, and leave others open. Sending it to the end of the list while hiding them stops the flow of my work, and makes it harder to organize my apps. The important thing to note is that this only happens when I had 2 apps open. When I had multiple apps, the hidden app just stayed in the same place. This did not happen when I had 2 apps open. Instead, the app went to the end of the list. The inconsistent behavior is the only reason that this is a bug report and not an enhancement.

**Screenshots / video**

![Screen Recording 2020-08-31 at 3 13 40 PM (1)](https://user-images.githubusercontent.com/30350506/91757781-e7868600-eb9c-11ea-80ee-8139a40cbc5b.gif)

**Steps to reproduce the bug**

1. Open multiple apps.
2. Hide first app
3. Notice that the app doesn't go to end of list.
4. Close one of the apps.
5. Try again and hide one of the apps.
6. Notice the behavior that takes place

**Comment by SoPat712** [comment]
After looking a bit more, I can see that this only occurs with more than 3 apps. I'm not sure why exactly that's happening, but then again I've never looked at the code once yet.

**Comment by lwouis** [comment]
Hi @SoPat712! Thank you for sharing this feedback!

This is not a bug. It's AltTab following the system events as they happen. I think you missed an important observation: if you hide the currently focused app, then macOS doesn't just hide its windows and keep it the active app in the menubar. No when you hide the currently focused app, its windows are hidden, and focus is switched to the app who owns the window now made visible.

It is this macOS interaction: the OS showing you the app underneath when you hide the current app, that is making the windows reorder in AltTab.

Now that I have explained that dynamic, what do you think?

**Comment by SoPat712** [comment]
I don't think I understand fully. Why does this interaction not take place with multiple apps? Shouldn't it also find the visible app then, and change the focus again as well?

**Comment by lwouis** [comment]
I can't tell from your gif because it's hard to follow what's happening with only seeing AltTab UI. What i can tell you is: when you hide an app which is active, the OS needs a new app to be active. Just hide the active app with macOS and see which name is shown in the top left of the screen.

Now if you hide an app that's not the active app, nothing extra happens. You stay on the active app you're in.

These, I think but i'm not sure cause i can't see enough information on an animated gif (a full screen video would help), are the 2 types of interactions you're experiencing i believe

**Comment by lwouis** [comment]
Here is a [video demonstration](https://www.loom.com/share/40cbb38a069a46d9b2b0d8deebb56520). Could you please have a look?

**Comment by SoPat712** [comment]
Oh, I completely understand now. Thanks for creating the visual to show me what was going on. I'll close the issue now then!
